crypto/CMakeLists.txt

```diff
@@ -110,12 +110,19 @@ else()
   elseif(APPLE)
     set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/mac-${ARCH}/crypto/)
   elseif(UNIX)
-    set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/linux-${ARCH}/crypto/)
+    if(${ARCH} STREQUAL "generic")
+      message(STATUS "Detected generic linux platform. No assembly files will be included.")
+    else()
+      set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/linux-${ARCH}/crypto/)
+    endif()
   elseif(WIN32)
     set(ASSEMBLY_SOURCE ${GENERATE_CODE_ROOT}/win-${ARCH}/crypto/)
   endif()
-  message("Copying platform assembly files from ${ASSEMBLY_SOURCE} to ${CMAKE_CURRENT_BINARY_DIR}")
-  file(COPY ${ASSEMBLY_SOURCE}/ DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+
+  if(DEFINED ASSEMBLY_SOURCE)
+    message("Copying platform assembly files from ${ASSEMBLY_SOURCE} to ${CMAKE_CURRENT_BINARY_DIR}")
+    file(COPY ${ASSEMBLY_SOURCE}/ DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+  endif()
```

Comment:
When it's generic but by default OPENSSL_NO_ASM is false, some c code may still try to look for assembly code (search the macro usage).

Should generic arch mean `OPENSSL_NO_ASM=1` because assembly is not generic? I guess :)
https://github.com/awslabs/aws-lc/blob/main/CMakeLists.txt#L585-L587

Reply:
You could do that I guess. But detecting ASM support should work independently of this.

Reply:
Compilation should fail if mips is not little-endian configured.